src/binary-encoding.adoc

```diff
@@ -0,0 +1,1618 @@
+[source]
+----
+┌──── The opcode `0xE6` indicates a variable-length sequence of symbol address annotations;
+│     a VarUInt follows representing the length of the sequence.
+│   ┌──── Annotations sequence length: VarUInt 3 with symbol address: VarUInt 10 ($10)
+│   │  ┌──── Annotation with symbol address: VarUInt 10 ($10)
+│   │  │  ┌──── Annotation with symbol address: VarUInt 11 ($11)
+│   │  │  │  ┌──── Annotation with symbol address: VarUInt 12 ($12)
+E5 07 15 17 19 5F
+               └── The annotated value: `false`
+----
+
+[[annotations_with_varsym_text]]
+==== Annotations with `VarSym` text
+
+Opcodes 0xE7 through 0xE9 indicate one or more annotations encoded as <<varsym, `VarSym`>>s.
+
+If the opcode is:
+
+* `0xE7`, a single `VarSym`-encoded symbol follows.
+* `0xE8`, two `VarSym`-encoded symbols follow.
+* `0xE9`, a `VarUInt` follows that represents the byte length of the annotations sequence, which is
+made up of any number of annotations encoded as ``VarSym``s.
+
+While this encoding is more flexible than <<annotations_with_symbol_addresses, annotations with
+symbol addresses>>, it can be slightly less compact when all the annotations are encoded as symbol
+addresses.
+
+==== Example encoding of `$10::false`
+[source]
+----
+┌──── The opcode `0xE7` indicates a single annotation encoded as a VarSym follows
+│  ┌──── Annotation with symbol address: VarSym 10 ($10)
+E7 15 5F
+      └── The annotated value: `false`
+----
+
+==== Example encoding of `foo::false`
+[source]
+----
+┌──── The opcode `0xE7` indicates a single annotation encoded as a VarSym follows
+│  ┌──── Annotation: VarSym -3; 3 bytes of UTF-8 text follow
+│  │   f  o  o
+E7 FD 66 6F 6F 5F
+      └──┬───┘ └── The annotated value: `false`
+      3 UTF-8
+       bytes
+----
+
+==== Example encoding of `$10::foo::false`
+
+Note that `VarSym` annotation sequences can switch between symbol address and inline text
+on a per-annotation basis.
+
+[source]
+----
+┌──── The opcode `0xE8` indicates two annotations encoded as VarSyms follow
+│  ┌──── Annotation: VarSym 10 ($10)
+│  │  ┌──── Annotation: VarSym -3; 3 bytes of UTF-8 text follow
+│  │  │   f  o  o
+E8 15 FD 66 6F 6F 5F
+         └──┬───┘ └── The annotated value: `false`
+         3 UTF-8
+          bytes
+----
+
+==== Example encoding of `$10::foo::$11::false`
+
+[source]
+----
+┌──── The opcode `0xE9` indicates a variable-length sequence of VarSym-encoded annotations
+│  ┌──── Length: VarUInt 6
+│  │  ┌──── Annotation: VarSym 10 ($10)
+│  │  │  ┌──── Annotation: VarSym -3; 3 bytes of UTF-8 text follow
+│  │  │  │           ┌──── Annotation: VarSym 11 ($11)
+│  │  │  │   f  o  o │
+E9 0D 15 FD 66 6F 6F 17 5F
+            └──┬───┘    └── The annotated value: `false`
+            3 UTF-8
+             bytes
+----
+
+[[nops]]
+=== ``NOP``s
+
+A `NOP` (short for "no-operation") is the binary equivalent of whitespace. `NOP` bytes have no meaning,
+but can be used as padding to achieve a desired alignment.
```

Comment:
I know the restrictions follow as 1.0, but it might be worth enumerating where `NOP` is allowed and how this works in the new flavors of struct.  I already saw the one error case of `NOP` after annotations--I kind of forgot if that was a 1.0 error too, so re-enumerating the validity is probably a good thing.

Reply:
Added.